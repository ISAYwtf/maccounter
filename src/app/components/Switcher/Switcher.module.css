.switch {
    position: relative;
    display: flex;
    --height-switch: 20px;
    height: var(--height-switch);
    width: calc(var(--height-switch) * 2);
}

.switch input {
    opacity: 0;
    height: 0;
    width: 0;
}

.slider {
    display: block;
    position: relative;
    cursor: pointer;
    height: var(--height-switch);
    width: calc(var(--height-switch) * 2);
    transition: 0.2s;
    background-color: var(--color-main);
    box-shadow: inset 0 0 0 1px var(--color-main);
    border-radius: var(--height-switch);
}

.slider:before {
    position: absolute;
    content: "";
    height: 16px;
    width: 16px;
    top: 50%;
    left: 2px;
    bottom: 2px;
    background-color: var(--color-bg-body);
    transform: translateY(-50%);
    border-radius: 50%;
    transition: 0.2s;
}

.switch input:checked + .slider {
    background-color: var(--color-bg-body);
}

.switch input:checked + .slider:before {
    transform: translateY(-50%) translateX(var(--height-switch));
    background-color: var(--color-main);
}
