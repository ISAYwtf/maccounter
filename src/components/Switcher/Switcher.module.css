.switch {
    position: relative;
    display: flex;
    height: 20px;
    width: 40px;
}

.switch input {
    opacity: 0;
    height: 0;
    width: 0;
}

.slider {
    display: block;
    position: relative;
    cursor: pointer;
    width: 100%;
    height: 100%;
    transition: 0.2s;
    background-color: var(--color-main);
    border: 1px solid var(--color-main);
    border-radius: 10px;
}

.slider:before {
    position: absolute;
    content: "";
    height: 16px;
    width: 16px;
    top: 50%;
    left: 2px;
    bottom: 2px;
    background-color: var(--color-bg-body);
    transform: translateY(-50%);
    border-radius: 50%;
    transition: 0.2s;
    /*box-shadow: 0 1px 1px var(--color-txt);*/
}

.switch input:checked + .slider {
    background-color: var(--color-bg-body);
}

.switch input:checked + .slider:before {
    transform: translateY(-50%) translateX(20px);
    background-color: var(--color-main);
}
