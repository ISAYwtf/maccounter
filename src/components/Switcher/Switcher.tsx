import React from "react"
import styles from "./Switcher.module.css"

const Switcher = () =>
    <label className={styles.switch}>
        <input type="checkbox"/>
        <span className={styles.slider}></span>
    </label>

export default Switcher
