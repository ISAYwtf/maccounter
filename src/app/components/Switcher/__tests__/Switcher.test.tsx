import React from 'react'
import { render } from '@testing-library/react'
import Switcher from '@components/Switcher/Switcher'

describe('Switcher', () => {
    const renderComponent = <Switcher />

    it('render Switcher', () => {
        const { asFragment } = render(renderComponent)
        expect(asFragment()).toMatchSnapshot()
    })
})
