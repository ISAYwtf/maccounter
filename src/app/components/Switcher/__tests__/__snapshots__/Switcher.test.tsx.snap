// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher render Switcher 1`] = `
<DocumentFragment>
  <label
    class="switch"
  >
    <input
      type="checkbox"
    />
    <span
      class="slider"
    />
  </label>
</DocumentFragment>
`;
